web/src/components/core/PasswordInput.test.jsx

```diff
@@ -0,0 +1,171 @@
+  });
+
+  it("can be disabled", () => {
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+        isDisabled={true}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+    const visibilityButton = screen.getByRole("button");
+
+    expect(inputField).toBeDisabled();
+    expect(visibilityButton).toBeDisabled();
+  });
+
+  it("applies autoFocus behavior correctly", () => {
+    plainRender(
+      <PasswordInput
+        autoFocus
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+
+    expect(document.activeElement).toBe(inputField);
+  });
+
+  it("displays helper text for invalid input", () => {
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+        helperTextInvalid="Invalid input"
+        validated="error"
+      />
+    );
+    const helperText = screen.getByText("Invalid input");
+
+    expect(helperText).toBeInTheDocument();
+  });
+
+  it("calls onBlur event handler", () => {
+    const onBlurMock = jest.fn();
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+        onBlur={onBlurMock}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+
+    fireEvent.blur(inputField);
+    expect(onBlurMock).toHaveBeenCalled();
+  });
+
+  it("ensures accessibility attributes are set correctly", () => {
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("Password")}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+    const visibilityButton = screen.getByLabelText("Password visibility button");
+
+    expect(inputField).toHaveAttribute("aria-label", "Password");
+    expect(visibilityButton).toHaveAttribute("aria-label", "Password visibility button");
+  });
+
+  it("handles setting and changing the input value", async () => {
+    let inputValue = "";
+    const handleChange = (value) => {
+      inputValue = value;
+    };
+
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("Password")}
+        value={inputValue}
+        onChange={handleChange}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+    
+    await fireEvent.change(inputField, { target: { value: "newpassword" }});
+    expect(inputValue).toBe("newpassword");
+  });
```

Comment:
Again, I think this is already tested by the components' library. But maybe I'm wrong, @imobachgs could you please double-check it?

Reply:
Thinking it twice, as `PasswordInput` is like _an internal library component_ it's ok testing it behaves as we expect. But I'd do it following the React Testing Library principles. How? By using a fake component in the test as a consumer of `PasswordInput`.

We've few examples already, but now I only see the [TestingPopup](https://github.com/openSUSE/agama/blob/2d4284241bbd3b0960df06f2a442f9d22dc6e74a/web/src/components/core/Popup.test.jsx#L33) one.

I'm in rush now, I will update the comment later with more examples.
...